pkg/grpc/registry/registry.proto

```diff
@@ -0,0 +1,82 @@
+syntax = "proto3";
+
+package registry;
+option go_package = "github.com/pomerium/pomerium/pkg/grpc/registry";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/any.proto";
+import "validate/validate.proto";
+
+enum ServiceKind {
+    UNDEFINED_DO_NOT_USE = 0;
+
+    // databroker grpc service
+    DATABROKER = 1;
+    // authorize grpc service
+    AUTHORIZE = 2;
+    // authentication http(s) service
+    AUTHENTICATE = 3;
+    // proxy service
+    PROXY = 4;
+    // registry service
+    REGISTRY = 5;
+    // console grpc service
+    CONSOLE = 6;
+    // prometheus metrics compatible http endpoint
+    PROMETHEUS_METRICS = 7;
+}
+
+message Service {
+    // kind is service kind
+    ServiceKind kind = 1 [(validate.rules).enum = {not_in: [0]}];
+    // network endpoints this service may be reachable at
+    string endpoint = 3 [(validate.rules).repeated.min_items = 1];
```

Comment:
What would "all-in-one" mode report as? All of them repeated?

Reply:
yes. however currently we only advertise metrics endpoint.